packages/services/src/services/Analytics/Analytics.js

```diff
@@ -267,14 +267,26 @@ class AnalyticsAPI {
       primaryCategory: 'VIDEO',
       eventName: data.title,
       eventCategoryGroup: data.playerType,
-      executionPath: data.videoId,
+      executionPath: data.videoId || data.mediaId,
       execPathReturnCode: playerState,
       eventVidStatus: data.playerState,
       eventVidTimeStamp: currentTime,
       eventVidLength: duration,
       eventVidPlayed: percentWatched + '%',
     };
 
+    if (data?.customMetricsData?.playerStateLabel) {
+      eventData.playerStateLabel = data.customMetricsData.playerStateLabel;
+    }
+
+    if (data?.customMetricsData?.driverId) {
+      eventData.driverId = data.customMetricsData.driverId;
+    }
+
+    if (data?.customMetricsData?.targetURL) {
+      eventData.targetURL = data.customMetricsData.targetURL;
+    }
```

Comment:
```suggestion
      driverId: data?.customMetricsData?.driverId,
      executionPath: data.videoId || data.mediaId,
      execPathReturnCode: playerState,
      eventVidStatus: data.playerState,
      eventVidTimeStamp: currentTime,
      eventVidLength: duration,
      eventVidPlayed: percentWatched + '%',
      playerStateLabel: data?.customMetricsData?.playerStateLabel,
      targetURL: data?.customMetricsData?.targetURL,
    };
```

would we be able to move the custom metrics data logic into the `eventData` declaration like this?

Reply:
@emyarod - As far as this was implemented on v18 (in last year) and as far as I understood, the 3 params (playerStateLabel, driverId and targetURL) were to be used exclusively with this event firing, not on all of then.

And by declaring the property on object initiating means that it will be fired as `undefined`, while having it under the above if statement, the object will have the property added only if the param is passed.